analysis/monkwindwalker/src/modules/covenants/FaelineStomp.tsx

```diff
@@ -0,0 +1,82 @@
+import { formatNumber, formatPercentage } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import { SpellIcon } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER, SELECTED_PLAYER_PET } from 'parser/core/Analyzer';
+import calculateEffectiveDamage from 'parser/core/calculateEffectiveDamage';
+import Events, { DamageEvent } from 'parser/core/Events';
+import Abilities from 'parser/core/modules/Abilities';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import React from 'react';
+import Enemies from 'parser/shared/modules/Enemies';
+
+
+
+const MOD = 0.08;
+
+class FaelineStompWindwalker extends Analyzer {
+
+  static dependencies = {
+    enemies: Enemies,
+  }
+  
+  protected enemies!: Enemies;
+  
+  totalDamage = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasLegendaryByBonusID(SPELLS.FAELINE_HARMONY.bonusID);
+    if (this.active) {
+      this.addEventListener(
+        Events.damage
+          .by(SELECTED_PLAYER | SELECTED_PLAYER_PET),
+        this.onAffectedDamage,
+      );
+    }
+  }
+  onAffectedDamage(event: DamageEvent) {
+    if (event.targetIsFriendly) {
+      return;
+    }
+    const target = this.enemies.getEntity(event);
+    if (target !== null && target.hasBuff(SPELLS.FAELINE_HARMONY_DEBUFF.id, event.timestamp)) {
```

Comment:
If I were to write this I would probably do const enemy = this.enemies.getEntity(event); 
if(!enemy || !enemy.hasBuff(...)) { return } 

to avoid potential nesting if statements in the future - but this is also fine however you can simplify it like this: 

```suggestion
    if (target && target.hasBuff(SPELLS.FAELINE_HARMONY_DEBUFF.id)) {
```